### PR TITLE
add annotations to ctfloat.d

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/root/ctfloat.d [hunk 1]
```diff
@@ -110,18 +113,23 @@ extern (C++) struct CTFloat
...
 
+    pure
     static real_t fmin(real_t x, real_t y) { return x < y ? x : y; }
+    pure
     static real_t fmax(real_t x, real_t y) { return x > y ? x : y; }
 
+    pure
     static real_t fma(real_t x, real_t y, real_t z) { return (x * y) + z; }
 
+    pure @trusted
```
Comment (jacob-carlborg) [inline src/dmd/root/ctfloat.d]: Are we putting the attributes on a separate line in other places? I don't think that is particular common to do in D.

Comment (n8sh) [reply src/dmd/root/ctfloat.d]: My assumption is the reason was to minimize the diff

Comment (WalterBright) [reply src/dmd/root/ctfloat.d]: that's right. It makes for easy review.

Comment (ZombineDev) [reply src/dmd/root/ctfloat.d]: @jacob-carlborg The style (attributes on the line before the declaration) is used in Phobos, as well as various other styles. Personally I prefer it, as it looks tidier and more elegant and is easier to keep up with the soft 80 columns rule. Also, I think it is consistent with the majority use of UDAs. Pragmatically speaking, one is allowed to use any style of attributes, until we add a rule to the style guide.

**Review by PetarKirov — APPROVED** [review]

**Comment by ZombineDev** [comment]
Auto-merge toggled on

**Pull request merged**